packages/typespec-client-generator-core/test/package.test.ts

```diff
@@ -2671,10 +2677,11 @@ describe("typespec-client-generator-core: package", () => {
       `
       );
       const sdkPackage = runnerWithCore.context.experimental_sdkPackage;
-      strictEqual(sdkPackage.clients.length, 2);
+      strictEqual(sdkPackage.clients.length, 1);
```

Comment:
curious why the client.length is 2 before in this case?

Reply:
see this:
```
@route("/widgets")
    @tag("Widgets")
    interface Widgets {
      ${code}
    }`);
    }
```

Reply:
we are only listing top-level clients in `.clients` now, that's the change we are making in this PR. There is still the interior client, it's just going to be listed as a subclient on the top-level client